fluss-connectors/fluss-connector-flink/src/main/java/com/alibaba/fluss/connector/flink/catalog/FlinkCatalog.java

```diff
@@ -359,15 +361,53 @@ public void alterTable(ObjectPath objectPath, CatalogBaseTable catalogBaseTable,
     @Override
     public List<CatalogPartitionSpec> listPartitions(ObjectPath objectPath)
             throws TableNotExistException, TableNotPartitionedException, CatalogException {
-        // TODO: use admin.listPartitionInfos()
-        return Collections.emptyList();
+
+        String tableName = objectPath.getObjectName();
+        TablePath tablePath;
+        try {
+            if (tableName.contains(LAKE_TABLE_SPLITTER)) {
+                tablePath =
+                        TablePath.of(
+                                objectPath.getDatabaseName(),
+                                tableName.split("\\" + LAKE_TABLE_SPLITTER)[0]);
+            } else {
+                tablePath = toTablePath(objectPath);
+            }
+            TableInfo tableInfo = admin.getTable(tablePath).get();
+            TableDescriptor tableDescriptor = tableInfo.getTableDescriptor();
+            if (!tableDescriptor.isPartitioned()) {
+                throw new TableNotPartitionedException(getName(), objectPath);
+            }
+            String partitionField = tableDescriptor.getPartitionKeys().get(0);
```

Comment:
why only get the first field partition key rather than all the partition keys?

Reply:
Currently, partitioned table only supports one partition key.